internal/api/attestationconfigapi/snp.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package attestationconfigapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/edgelesssys/constellation/v2/internal/attestation/variant"
+)
+
+// AttestationURLPath is the URL path to the attestation versions.
+const AttestationURLPath = "constellation/v1/attestation"
+
+// SEVSNPVersion tracks the latest version of each component of the Azure SEVSNP.
+type SEVSNPVersion struct {
+	// Bootloader is the latest version of the Azure SEVSNP bootloader.
+	Bootloader uint8 `json:"bootloader"`
+	// TEE is the latest version of the Azure SEVSNP TEE.
+	TEE uint8 `json:"tee"`
+	// SNP is the latest version of the Azure SEVSNP SNP.
+	SNP uint8 `json:"snp"`
+	// Microcode is the latest version of the Azure SEVSNP microcode.
+	Microcode uint8 `json:"microcode"`
+}
+
+// SEVSNPVersionAPI is the request to get the version information of the specific version in the config api.
+// Because variant is not part of the marshalled JSON, fetcher and client methods need to fill the variant property.
+// Once we switch to v2 of the API we should embed the variant in the object.
```

Comment:
I am not sure I understand the question but these two facts made me write the comment:
- we can't add a property to the JSONs in the API within one version because old clients will break. this is why we can't put the variant into the JSON in API version 1.
- we want the variant in the JSON so `client.Fetch(..)` returns a SEVSNPVersionAPI object that has a valid `Variant` property set.